File: Crio/Session.cpp
#include <bits/stdc++.h>
using namespace std;

long long connectRopes(vector<long long > arr)
{
    if(arr.size() == 0 or arr.size() == 1){ // case where there is an empty arr
        return 0;
    }
    long long total = 0;
    priority_queue < long long , vector<long long>, greater <long long> > pq;
    for (int i = 0; i < arr.size(); i++){
        pq.push(arr[i]);    // this will have the shortest length at the top
    }
    // the idea is to keep on adding the two smallest lengths to minimize the cost 
    // we use heaps to keep track of the smallest element all the time 
    while(pq.size() != 1){
        long long f_min = pq.top();
        pq.pop();
        long long s_min = pq.top();
        pq.pop();
        long long sum = f_min + s_min;
        pq.push(sum);  // add it back into the min heap as it needs to be added with the other lengths as well.
        total += sum;
    }
    return total;
}

int main()
{
    int n;
    cin>>n;
    vector<long long > arr(n);
    for(int i=0;i<n;i++)
        cin>>arr[i];
    long long result = connectRopes(arr);
    cout<<result<<"\n";
    return 0;
}

File: Crio/Session.java
import java.util.*;

class ConnectRopes{

	private static long connectRopes(Vector<Long> arr)
	{
		long result = 0;
		PriorityQueue<Long> heap = new PriorityQueue<Long>();
		for (int i =0;i<arr.size();i++)
		{
			heap.add(arr.get(i));
		} 
		while(heap.size()>=2)
		{
			long first = heap.poll();
			long second = heap.poll();
			long sum = first+second;
			heap.add(sum);
			result = result+sum;
		}
		
		return result;
	}	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		Vector<Long> arr = new Vector<Long>();
		for(int i=0;i<n;i++)
			arr.add(sc.nextLong());
		System.out.println(connectRopes(arr));
	}
}
